### PR TITLE
Add package descriptions to Dokka

### PR DESCRIPTION
**Pull request by rnicoll**
Add package descriptions used by Dokka

Part of https://r3-cev.atlassian.net/browse/CORDA-499

### CODE REVIEW

**Comment by rnicoll** [comment]
Primarily I'd appreciate thoughts on any packages I've misunderstood intention of.

File: docs/packages.md [hunk 1]
```diff
@@ -1,3 +1,84 @@
+# Package net.corda.client.jfx.model
+
+Data models for the JavaFX client to Corda. These contain observables which are updated via the Corda RPC interface and notify JavaFX of changes.
```
Comment (mikehearn) [inline docs/packages.md]: These are also intended for application developers to use in their own GUI apps. *Every* description should be written from the perspective of our users. If something exists only for us, it shouldn't appear in the javadocs at all.

File: docs/packages.md [hunk 2]
```diff
@@ -1,3 +1,84 @@
...
+
+Tools used by the client to produce mock data for testing purposes.
+
+# Package net.corda.client.rpc
+
+RPC client interface to Corda, for use both by user-facing client and integration with external systems.
+
+# Package net.corda.core.concurrent
+
+Concurrency/multi-threading support classes for Corda, primarily providing support for the promises pattern.
```
Comment (mikehearn) [inline docs/packages.md]: Let's just change this sentence to "Provides a simplified Future class that allows registration of a callback to run when the future is complete." ... as that's all this package has right now.

File: docs/packages.md [hunk 3]
```diff
@@ -1,3 +1,84 @@
...
+
+See [FlowLogic] for the basic class all flows extend, or [CollectSignaturesFlow] for a simple example flow. Flows are started via a node's service hub.
+
+# Package net.corda.core.identity
+
+Data classes which model different forms of identity (potentially with supporting evidence) for legal entities and services.
+
+# Package net.corda.core.internal
+
+APIs and implementations which are currently being revised and are subject to future change. Generally these should be
```
Comment (mikehearn) [inline docs/packages.md]: Let's simplify this to: "Internal, do not use" for all internal packages. And we should find a way to suppress them in the output.

File: docs/packages.md [hunk 4]
```diff
@@ -1,3 +1,84 @@
...
+
+Base data types for smarts contracts implemented in Corda. To implement a new contract start with [Contract], or see the examples in [net.corda.finance.contracts].
+
+# Package net.corda.core.crypto
+
+Cryptography data and utility classes used for signing, verifying, key management and data integrity checks.
+
+# Package net.corda.core.crypto.composite
+
+Composite key and signature classes, which are used to represent the signing requirements for multisignature scenarios such as RAFT notary services.
```
Comment (mikehearn) [inline docs/packages.md]: I suspect we should combine crypto and crypto.composite

File: docs/packages.md [hunk 5]
```diff
@@ -1,3 +1,84 @@
...
+Data classes which model different forms of identity (potentially with supporting evidence) for legal entities and services.
+
+# Package net.corda.core.internal
+
+APIs and implementations which are currently being revised and are subject to future change. Generally these should be
+treated as non-public.
+
+# Package net.corda.core.jackson
+
+Support classes for integrating the Jackson JSON serializer/deserializer with Corda.
```
Comment (mikehearn) [inline docs/packages.md]: Shouldn't we move this package under net.corda.client? It's in the client module, no?

File: docs/packages.md [hunk 6]
```diff
@@ -1,3 +1,84 @@
...
+
+# Package net.corda.core.node.services
+
+Services which run within a Corda node and provide various pieces of functionality such as identity management, transaction storage, etc.
+
+# Package net.corda.core.node.services.vault
+
+Supporting data types for the vault services.
```
Comment (mikehearn) [inline docs/packages.md]: Again, I wonder if we need a separate package for this.

File: docs/packages.md [hunk 7]
```diff
@@ -1,3 +1,84 @@
...
+A collection of utilities for summing financial states, for example, summing obligations to get total debts.
+
+# Package net.corda.node.internal
+
+Node APIs and implementations which are currently being revised and are subject to future change. Generally these should be
+treated as non-public.
+
+# Package net.corda.node.services
+
+Implementations of services used by Corda nodes.
```
Comment (mikehearn) [inline docs/packages.md]: Packages under net.corda.node shouldn't have javadocs being emitted. Perhaps we need to adjust the arguments to Dokka.

File: docs/packages.md (hunk 5 above)
Comment (rnicoll) [reply docs/packages.md]: Some of my package docs may in fact be passive-aggressive comments on the structure, yes :-D

Comment (rnicoll) [reply docs/packages.md]: Oh, apologies, this one was actually more a "Do we really need a separate package" thought, and having seen how the code is structured, we do. I'll update the package docs to talk about why.

File: docs/packages.md [hunk 8]
```diff
@@ -1,3 +1,84 @@
...
+
+Internal, do not use. These APIs and implementations which are currently being revised and are subject to future change.
+
+# Package net.corda.node.services
+
+Implementations of services used by Corda nodes.
+
+# Package net.corda.node.services.api
+
+Non-core node service APIs; typically these are not used outside of the node itself.
```
Comment (mnesbit) [inline docs/packages.md]: Really this is meant to be internal interfaces to allow alternate implementations of node components to inter-operate within the node.

File: docs/packages.md [hunk 9]
```diff
@@ -1,3 +1,84 @@
...
+
 # Package net.corda.finance.utils
 
-A collection of utilities for summing financial states, for example, summing obligations to get total debts.
\ No newline at end of file
+A collection of utilities for summing financial states, for example, summing obligations to get total debts.
+
+# Package net.corda.node.internal
+
+Internal, do not use. These APIs and implementations which are currently being revised and are subject to future change.
```
Comment (mnesbit) [inline docs/packages.md]: In practice EVERYTHING in node is internal and implementation may change. This folder name is really a bit of a misnomer as these are the bootstrap classes and some people may need to extend from AbstractNode.

File: docs/packages.md [hunk 10]
```diff
@@ -1,3 +1,84 @@
...
+
+# Package net.corda.core.transactions
+
+Base data types for transactions which modify contract state on the distributed ledger.
+
+The core transaction on the ledger is [WireTransaction], which is constructed by [TransactionBuilder]. Once signed a transaction is stored
+in [SignedTransaction] which encapsulates [WireTransaction]. Finally there is a special-case [LedgerTransaction] which is used by contracts
+validating transactions, and is built from the wire transaction by resolving all references into their underlying data (i.e. inputs are
+actual states rather than state references).
+
```
Comment (mnesbit) [inline docs/packages.md]: The finance module is a cordapp and should be linked against as using our gradle plugins specific dependency types i.e. cordapp/testCompile for other custom cordapps and compile only when used for RPC clients.

Comment (rnicoll) [reply docs/packages.md]: Moving finance into a pure CorDapp is targeted for post-1.0 isn't it? Should I just remove them from the docs for now and tackle later?

**Comment by rnicoll** [comment]
@mikehearn thoughts on getting this merged? Will continue revising over time, obviously, this is just a first batch done

File: docs/packages.md (hunk 10 above)
Comment (mnesbit) [reply docs/packages.md]: Already on 1.0

**Pull request merged**